cpp/mysql_database_patcher.cc

```diff
@@ -161,8 +161,10 @@ int Main(int argc, char *argv[]) {
         std::string database;
         unsigned version;
         SplitIntoDatabaseAndVersion(update_filename, &database, &version);
-        if (last_schema.empty() or database != last_schema)
-            db_connection.queryOrDie("USE " + database);
+        if (last_schema.empty() or database != last_schema) {
+            if (db_connection.mySQLDatabaseExists(database))
```

Comment:
There is already existing mySQLDatabaseExists logic inside the ApplyUpdate function, so please move the USE statement there.